src/main/java/ru/sbt/mipt/oop/EventHandlingLoop.java

```diff
@@ -0,0 +1,86 @@
+package ru.sbt.mipt.oop;
+
+import static ru.sbt.mipt.oop.SensorEventType.*;
+
+public class EventHandlingLoop {
+
+    public static void handleEvents(SmartHome smartHome) {
+        // начинаем цикл обработки событий
+        while (true) {
+            SensorEvent event = getNextSensorEvent();
+            if (event == null) {
+                break;
+            }
+            System.out.println("Got event: " + event);
+            if (event.getType() == LIGHT_ON || event.getType() == LIGHT_OFF) {
+                // событие от источника света
+                handleLightEvent(event, smartHome);
+            }
+            if (event.getType() == DOOR_OPEN || event.getType() == DOOR_CLOSED) {
+                // событие от двери
+                handleDoorEvent(event, smartHome);
+            }
+        }
+    }
+
+    private static void handleLightEvent(SensorEvent event, SmartHome smartHome) {
+        for (Room room : smartHome.getRooms()) {
+            for (Light light : room.getLights()) {
+                if (light.getId().equals(event.getObjectId())) {
+                    boolean isSetLightOn = (event.getType() == LIGHT_ON);
+                    light.setOn(isSetLightOn);
+                    String stringToPrint = "Light " + light.getId() + " in room " + room.getName() + " was turned ";
+                    if (isSetLightOn) {
+                        System.out.println(stringToPrint + "on.");
+                    } else {
+                        System.out.println(stringToPrint + "off.");
+                    }
+                }
+            }
+        }
+    }
+
+    private static void handleDoorEvent(SensorEvent event, SmartHome smartHome) {
+        for (Room room : smartHome.getRooms()) {
+            for (Door door : room.getDoors()) {
+                if (door.getId().equals(event.getObjectId())) {
+                    boolean isSetDoorOpen = (event.getType() == DOOR_OPEN);
+                    door.setOpen(isSetDoorOpen);
+                    String stringToPrint = "Door " + door.getId() + " in room " + room.getName() + " was ";
+                    if (isSetDoorOpen) {
+                        System.out.println(stringToPrint + "opened.");
+                    } else {
+                        System.out.println(stringToPrint + "closed.");
+                    }
+                    additionalActionInCertainRoomsOnDoorEvent(event, room, smartHome);
```

Comment:
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D1%80%D0%B0%D0%B7%D0%B4%D0%B5%D0%BB%D1%8F%D0%B9%D1%82%D0%B5-%D0%B8%D0%B7%D0%BC%D0%B5%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5-%D0%BE%D0%B1%D1%8A%D0%B5%D0%BA%D1%82%D0%BE%D0%B2-%D0%B8-%D1%81%D1%86%D0%B5%D0%BD%D0%B0%D1%80%D0%B8%D0%B8-%D1%83%D0%BC%D0%BD%D0%BE%D0%B3%D0%BE-%D0%B4%D0%BE%D0%BC%D0%B0